docs/future/vite.md

```diff
@@ -615,6 +615,43 @@ const posts = import.meta.glob("./posts/*.mdx", {
 });
 ```
 
+#### Strict route exports
+
+With Vite, Remix gets stricter about which exports are allowed from your route modules.
+Previously, the Remix compiler would allow any export from routes.
+While this was convenient, it was also a common source of bugs that were hard to track down because they only surfaced at runtime.
```

Comment:
However, this approach doesn't work in Vite because it processes each module in isolation during development without any tree shaking. This means that any exports that were only intended for use on the server will also end up on the client, leading to errors or other unexpected behavior.